gradio/components/login_button.py

```diff
@@ -0,0 +1,89 @@
+"""Predefined button to sign in with Hugging Face in a Gradio Space."""
+from __future__ import annotations
+
+from typing import Any, Literal
+
+from gradio_client.documentation import document, set_documentation_group
+
+from gradio.components import Button
+from gradio.routes import Request
+
+set_documentation_group("component")
+
+
+@document()
+class LoginButton(Button):
+    """
+    Button that redirects the user to Sign with Hugging Face using OAuth.
+    """
+
+    is_template = True
+
+    def __init__(
+        self,
+        *,
+        value: str = "Sign in with Hugging Face",
+        variant: Literal["primary", "secondary", "stop"] = "secondary",
+        size: Literal["sm", "lg"] | None = None,
+        icon: str | None = "https://huggingface.co/front/assets/huggingface_logo-noborder.svg",
+        link: str | None = None,
+        visible: bool = True,
+        interactive: bool = True,
+        elem_id: str | None = None,
+        elem_classes: list[str] | str | None = None,
+        scale: int | None = 0,
+        min_width: int | None = None,
+        _activate: bool = True,
+        **kwargs,
+    ):
+        super().__init__(
+            value,
+            variant=variant,
+            size=size,
+            icon=icon,
+            link=link,
+            visible=visible,
+            interactive=interactive,
+            elem_id=elem_id,
+            elem_classes=elem_classes,
+            scale=scale,
+            min_width=min_width,
+            **kwargs,
+        )
+        if _activate:
+            self.activate()
+
+    def activate(self):
+        # Taken from https://cmgdo.com/external-link-in-gradio-button/
+        # Taking `self` as input to check if user is logged in
```

Comment:
Clever, though maybe a bit fragile. Until we can support reading `State` variables in the frontend (#3525), this is fine.